### ISSUE TITLE
yolov5.onnx view on netron

### ISSUE THREAD

**Issue by ashwinimate** [ultralytics/yolov5#13584]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

hello sir/mam,
 i have downloaded the yolov5 ptretrain weights for v6.0 (yolo5s.pt) using folowing comnad

!wget https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5s.pt

and exported to onnx file. While visualizing  the backbone architecture on netron, the number of bottleneck in C3 should be 3, 6 9, in order. howeever upon observing the number of bollteneck in the C3 block of backbone that i got from are 1,2,3,1 respectivly. Could you please help me to find where i can get the correct pretrained weight for yolov5 (all models)? have i downloaded the wring weights.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ashwinimate, thank you for reaching out and using YOLOv5 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your question.

Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for step-by-step guides, including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** (MRE) showing your export process (code snippets, commands, and model files used). This will help us debug the issue more efficiently.

If you have a custom training ❓ Question or are looking for specific pretrained weights, please share as much detail as possible, such as the exact model version, download source, and steps taken. Also confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements

[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will assist you soon! If you can, please update your issue with a reproducible example of your export workflow and clarify the exact model version and files you are comparing. Thank you for your patience and for helping improve YOLOv5! 😊

**Comment by ashwinimate** [comment]
as seen in the following image ther eis only one botttleneck in the c3 block in backbone. 
it should have been 3 bolltneck.

![Image](https://github.com/user-attachments/assets/ce855a30-d94b-42ce-b4e8-aa79e3017d53)

![Image](https://github.com/user-attachments/assets/f958bb22-2010-412e-8f10-7580e3fe2267)

**Comment by pderrenger** [comment]
Hello @ashwinimate,

The bottleneck counts you're observing in Netron (1, 2, 3, 1) are actually correct for the YOLOv5s v6.0 model. Each YOLOv5 size variant has a different number of bottlenecks in its C3 blocks:

- YOLOv5s: 1, 2, 3, 1 bottlenecks
- YOLOv5m: 2, 4, 6, 2 bottlenecks  
- YOLOv5l: 3, 6, 9, 3 bottlenecks
- YOLOv5x: 4, 8, 12, 4 bottlenecks

The "3, 6, 9" pattern you're referring to matches the YOLOv5l model, not YOLOv5s. Your download and export process was correct, and you have the proper weights for YOLOv5s v6.0.

**Comment by ashwinimate** [comment]
Thank you so much for clearing my doubt. i have one mor query though. when i see model in yolov5l.yaml, yolov5s.yaml,yolov5m.yaml,yolov5n.yaml i get the folowing for back bone archtitecture

# YOLOv5 v6.0 backbone
backbone:
  # [from, number, module, args]
  [
    [-1, 1, Conv, [64, 6, 2, 2]], # 0-P1/2
    [-1, 1, Conv, [128, 3, 2]], # 1-P2/4
    [-1, **3**, C3, [128]],
    [-1, 1, Conv, [256, 3, 2]], # 3-P3/8
    [-1, **6**, C3, [256]],
    [-1, 1, Conv, [512, 3, 2]], # 5-P4/16
    [-1, **9**, C3, [512]],
    [-1, 1, Conv, [1024, 3, 2]], # 7-P5/32
    [-1, **3**, C3, [1024]],
    [-1, 1, SPPF, [1024, 5]], # 9
  ]

from above the numver of bottleneck in C3 block are 3,6,9,3 (in bolde). if all the yaml file contain same backbone  forl all the sacl of model then how the netron visualization is showing different battleneck in differentr scale.

**Comment by ashwinimate** [comment]
please clear my doubt. looking forward for your responce. 
thank you

**Comment by pderrenger** [comment]
Hello @ashwinimate,

Thank you for your follow-up question! You're right that there appears to be a discrepancy between the YAML files and what you're seeing in Netron.

This is because YOLOv5 applies a depth multiplier to scale the number of bottlenecks based on model size. The YAML files show the base architecture, but during model construction, the actual number of bottlenecks is adjusted by multiplying the values in the YAML by a depth factor:

- YOLOv5n: 0.33× depth factor (1, 2, 3, 1 becomes approximately 0, 1, 1, 0)
- YOLOv5s: 0.33× depth factor (1, 2, 3, 1)
- YOLOv5m: 0.67× depth factor (2, 4, 6, 2)
- YOLOv5l: 1.0× depth factor (3, 6, 9, 3) - this is the base YAML you're seeing
- YOLOv5x: 1.33× depth factor (4, 8, 12, 4)

So the 3, 6, 9, 3 in the YAML represents the full-scale YOLOv5l model, while the smaller models like YOLOv5s have proportionally fewer bottlenecks. Your Netron visualization showing 1, 2, 3, 1 for YOLOv5s is absolutely correct.

**Comment by ashwinimate** [comment]
Thank You so much for the clarification. It was very helpfull.

**Comment by pderrenger** [comment]
You're welcome, @ashwinimate! I'm glad I could help clarify how the depth scaling works across the different YOLOv5 model variants. If you have any other questions about YOLOv5 in the future, feel free to ask. Happy modeling!
